assets/js/assessment_v2.js

```diff
@@ -106,80 +106,53 @@ function updateQuestions() {
-
-      // add to the number of correct answers
+      // if answer is correct
       numCorrect++;
-
-      // color the answers green
-      //answerContainers[questionNumber].style.color = "lightgreen";
-      // Show all explanations
-      if (currentQuestion.explanations) {
-        for (let answer in currentQuestion.answers) {
-          let explanation = currentQuestion.explanations[answer];
-          let explanationButton = document.getElementById(
-            "explanation" + questionNumber.toString() + answer
-          );
-          if (explanation) {
-            explanationButton.parentElement.nextElementSibling.innerHTML = explanation;
-            explanationButton.style.display = "inline-block";
-          } else {
-            explanationButton.style.display = "none";
-          }
-        }
-      }
+      document.getElementById(
+        "answer" + questionNumber.toString() + userAnswer
+      ).style.color = "lightgreen";
+      //console.log(`Question ${questionNumber}: Correct answer selected. Color changed to lightgreen.`);
     } else if (userAnswer) {
-      // if answer is wrong or blank
-      // color the answers red
+      // if answer is wrong
       document.getElementById(
         "answer" + questionNumber.toString() + userAnswer
       ).style.color = "red";
-      // Show only explanation for wrong answer
-      if (currentQuestion.explanations && userAnswer) {
-        let explanation = currentQuestion.explanations[userAnswer];
-        let explanationButton = document.getElementById(
-          "explanation" + questionNumber.toString() + userAnswer
-        );
-        if (explanation) {
-          explanationButton.parentElement.nextElementSibling.innerHTML = explanation;
-          explanationButton.style.display = "inline-block";
-        } else {
-          explanationButton.style.display = "none";
-        }
+      //console.log(`Question ${questionNumber}: Incorrect answer selected. Color changed to red.`);
+    }
+
+    // Show explanation for the selected answer
+    if (currentQuestion.explanations && userAnswer) {
+      let explanation = currentQuestion.explanations[userAnswer];
+      let explanationButton = document.getElementById(
+        "explanation" + questionNumber.toString() + userAnswer
+      );
+      if (explanation) {
+        explanationButton.parentElement.nextElementSibling.innerHTML = explanation;
+        explanationButton.style.display = "inline-block";
+      } else {
+        explanationButton.style.display = "none";
       }
     }
   });
+
   // show number of correct answers out of total
-  resultsContainer.innerHTML = `Score: ${numCorrect} out of ${toatNum}`;
+  resultsContainer.innerHTML = `Score: ${numCorrect} out of ${totalNum}`;
+  //console.log(`Results displayed: ${numCorrect} out of ${totalNum}`);
 }
 
 populateQuestions();
 addEventListener_explanations();
 addEventListener_checkbox();
 submitButton.addEventListener("click", showResults);
+
+populateQuestions();
```

Comment:
Why have calls to these functions been repeated?

Reply:
yes, it was a mistake I might have made while merging.